QHotkey/qhotkey_x11.cpp

```diff
@@ -1,7 +1,21 @@
 #include "qhotkey.h"
 #include "qhotkey_p.h"
-#include <QDebug>
-#include <QX11Info>
+
+#if QT_VERSION >= QT_VERSION_CHECK(6, 2, 0)
+	#include <QGuiApplication>
+	
+	#define _NATIVE_EVENT_RESULT qintptr
+	#define _OVERRIDE_IDENTIFIER override
+#elif QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+	#error "Qt 6.2.0 or greater is required when using Qt6"
+#else
+	#include <QDebug>
+	#include <QX11Info>
+	
+	#define _NATIVE_EVENT_RESULT long
+	#define _OVERRIDE_IDENTIFIER Q_DECL_OVERRIDE
```

Comment:
Maybe just we get rid of this old stuff and use `override` for both version? `Q_DECL_OVERRIDE` is for pre C++11 :)

Reply:
This was also my impression, but I don't wanted to modify the Qt5 code to avoid any problems. The Qt5 code is completely unchanged on my pull request.

There are also other old things that can be changed on the code. For example, we are still using Qt3 (yes, Qt3, not a typo) code in some parts, like [here](https://github.com/Skycoder42/QHotkey/blob/eb7ddaba57b108f99c2fefcd4eb23cef9c0ed681/QHotkey/qhotkey.cpp#L14-L15). `Qt::KeyboardModifierMask` is a member of the `Qt::ButtonState` enum, which was depracated in the Qt4 days. I was impressed to see that there is no compiler deprecation warning about this, even on Qt6. As I said, I tried to be as conservative as I could when doing the Qt6 changes.

But as you wish :)

Please confirm if you want me to push the `override` change.

Reply:
> Qt::KeyboardModifierMask is a member of the Qt::ButtonState enum, which was depracated in the Qt4 days.

That's interesting. I would happy to replace it, but let's leave it for another PR.

> I tried to be as conservative as I could when doing the Qt6 changes.

Totally understand! But let's drop `Q_DECL_OVERRIDE` to avoid extra macro :) Pretty sure that nobody will use this library without C++11.

Reply:
Nice. I'll push the override change, thanks.

I also think it's better to separate the work on deprecated parts to another pull request. I have other suggestions to merge, more specifically on the CMake code, and I'm planning to open more pulls if this one get merged :D

Reply:
It would be great!